### PR TITLE
editor/code/test: lay the foundation for writing tests

### PR DESCRIPTION
**Pull request by artagnon**
We have picked the Jasmine testing framework, and included a test for extension activation. Unfortunately, there seems to be some misunderstanding with Jasmine Spies, as they don't seem to capture the activation information message.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,12 +2,6 @@ _build
...
 *.install
-log-*.txt
-
 .git-ps
 
 *.cache
-
-editor/code/out
-
-result
```
Comment (Alizter) [inline .gitignore:13]: This gets produced by `nix build`.

Comment (artagnon) [reply .gitignore:13]: This belongs to the commit in #171. Only the last commit is the subject of this PR.

**Comment by Alizter** [comment]
It would be good to resurrect this. @artagnon could you list what is left to be done other than rebase?

**Comment by artagnon** [comment]
I think this is the wrong approach, and that we should pick `wdio-vscode` as the testing framework.

**Comment by artagnon** [comment]
Actually, I'm not sure anymore. @ejgallego pointed me to https://github.com/coq-community/vscoq/pull/372, which is where I got the idea for `wdio-vscode`. Unfortunately, that PR seems to be closed, citing reasons of flakiness.

Writing tests independently of vscode is another alternative, but after some digging, it seems that the only full-featured LSP clients are those that ship with editors; there seems to be no real LSP client written in Typescript independent of VSCode APIs, using the client portion of ocaml-lsp-server will blow up the size of our tests, including lots of irrelevant details like YoJson parsing. I will continue investigating.

**Pull request closed without merging**